pydantic/config.py

```diff
@@ -233,9 +267,10 @@ def build_config(
     return new_model_config
 
 
-def prepare_config(config: ConfigDict, cls_name: str) -> None:
-    if not isinstance(config['extra'], Extra):
+def prepare_config(config: ConfigDict | dict[str, Any], error_label: str) -> None:
+    extra = config.get('extra')
+    if extra is not None and not isinstance(extra, Extra):
         try:
             config['extra'] = Extra(config['extra'])
         except ValueError:
-            raise ValueError(f'"{cls_name}": {config["extra"]} is not a valid value for "extra"')
+            raise ValueError(f'{error_label!r}: {config["extra"]!r} is not a valid value for config[{"extra"!r}]')
```

Comment:
I felt this name was more appropriate now that it can come from non-classname sources.